beacon-chain/db/kv/slasher.go

```diff
@@ -197,6 +197,54 @@ func (s *Store) SaveSlasherChunks(
 	})
 }
 
+// CheckAndUpdateForSlashableProposals takes in a list of proposals and for each,
+// checks if there already exists a proposal at the same slot+validatorIndex combination. If so,
+// we append that existing proposal to a list. Otherwise, we save the proposal to disk, then append nil
+// to a list. We then return the list to the caller.
+func (s *Store) CheckAndUpdateForSlashableProposals(
```

Comment:
`FooAndBar` function names usually are a sign of a code smell, implying that the function should be broken into two smaller functions. Do you think it's possible here?